src/qt3utils/datagenerators/spectrometers/spectrometer.py

```diff
@@ -0,0 +1,230 @@
+    """
+    Filters out kwargs that are not valid for the given method.
+    """
+    return {k: v for k, v in all_kwargs.items() if k in get_method_argument_names(method)}
+
+
+class Spectrometer(abc.ABC):
+
+    DEVICE_NAME: str = ''
+    """ The name of the device used for logging purposes. """
+
+    ACQUISITION_MODES: set[str] = {'single', 'step-and-glue', 'kinetic series', 'accumulation'}
+    """ The supported acquisition modes. """
+
+    def __init__(self):
+        logging.basicConfig(level=logging.INFO)
+        self.logger = logging.getLogger(self.DEVICE_NAME if self.DEVICE_NAME != '' else self.__class__.__name__)
+
+    @abc.abstractmethod
+    def open(self) -> None:
+        """
+        Initializes the connection to devices.
+        """
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    def close(self) -> None:
+        """
+        Terminates connection to devices.
+        """
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    @property
+    def grating_list(self) -> List[str]:
+        """
+        A list of all installed gratings.
+        """
+        return []
+
+    @abc.abstractmethod
+    @property
+    def current_grating(self) -> str:
+        """
+        Current spectrometer grating.
+        """
+        return ''
+
+    @abc.abstractmethod
+    @current_grating.setter
+    def current_grating(self, value: str) -> None:
+        """
+        Set the spectrometer grating.
+        """
+        pass
+
+    @abc.abstractmethod
+    @property
+    def center_wavelength(self) -> float:
+        """
+        The grating center wavelength.
+        """
+        return np.nan
+
+    @abc.abstractmethod
+    @center_wavelength.setter
+    def center_wavelength(self, nanometers: float) -> None:
+        """
+        Sets the grating center wavelength.
+        """
+        pass
+
+    @abc.abstractmethod
+    def get_wavelengths(self) -> np.ndarray:
+        """
+        Returns the wavelength calibration for a single frame.
+        """
+        pass
+
+    @abc.abstractmethod
+    @property
+    def sensor_temperature_set_point(self) -> float:
+        """
+        The sensor set-point temperature in Celsius.
+        """
+        return np.nan
+
+    @abc.abstractmethod
+    @sensor_temperature_set_point.setter
+    def sensor_temperature_set_point(self, deg_celsius: float) -> None:
+        """
+        Sets the sensor target temperature in Celsius.
+        """
+        pass
+
+    @abc.abstractmethod
+    @property
+    def exposure_time(self) -> float:
+        """
+        Returns the single frame exposure time (in ms).
+        """
+        return np.nan
+
+    @abc.abstractmethod
+    @exposure_time.setter
+    def exposure_time(self, ms: float) -> None:
+        """
+        Sets the single frame exposure time in milliseconds.
+        """
+        pass
+
+    @property
+    def clock_rate(self) -> float:
```

Comment:
Thinking about it now. Why not just have this in the controller? What is its relation with the spectrometer settings? AFAIK this is not something we find in spectrometer software and would be make more sense in each persons individual spectrometer controller. Addiitonally im curious where this is being used